command/operator_migrate.go

```diff
@@ -365,10 +386,24 @@ func parseStorage(result *migratorConfig, list *ast.ObjectList, name string) err
 
 // dfsScan will invoke cb with every key from source.
 // Keys will be traversed in lexicographic, depth-first order.
-func dfsScan(ctx context.Context, source physical.Backend, cb func(ctx context.Context, path string) error) error {
+func dfsScan(ctx context.Context, source physical.Backend, maxParallel int, cb func(ctx context.Context, path string) error) error {
 	dfs := []string{""}
+	if maxParallel < 1 {
+		maxParallel = 1
+	}
+
+	permitPool := physical.NewPermitPool(maxParallel)
+	eg, ctx := errgroup.WithContext(ctx)
```

Comment:
I wonder if both `errgroup` and `permitPool` are necessary? It seems that both are doing the same thing except `errgroup` is doing error propagation as well. 
Check the new line 424, both `eg.Go` and `permitPool.Acquire` are used together.